### DISCUSSION TITLE
High disk usage with data importing and exporting

### DISCUSSION THREAD

**Discussion by rongxinyu** [duckdb/duckdb#5785] [Q&A]
Hi, 

I am trying out duckdb and amazed by its performance. However I ran into one issue. I have a csv.gz file and when it's loaded into duckdb, it's about 7 gigabytes.

When I load the csv into duckdb and try to export the table into a parquet file, I noticed that the disk usage can easily go up to 70 gigabytes, most of which goes into the temp folder.

I wonder what's going on and how to reduce the disk usage. 

Thanks

**Comment by l1t1** [comment]
add memory?

**Reply by rongxinyu** [reply]
Thanks for the suggestion. I set the memory limit to 26G as my RAM is 32G. Wasn't that enough?

**Comment by l1t1** [comment]
Or you can try other tools
https://stackoverflow.com/questions/50604133/convert-csv-to-parquet-file-using-python

**Reply by rongxinyu** [reply]
Thanks. This is what I did in the end, converting it from CSV to parquet with pyarrow.

I am still interested in importing parquet data into duckdb as with smaller datasets, querying against duckdb format is faster than parquet.

I did a rough estimation that the data will be around 50Gs without compression, and that might explain the large disk usage. I am still curious about how the data import works on a high level.
